x-pack/plugins/spaces/server/config.ts

```diff
@@ -27,9 +28,19 @@ export function createConfig$(context: PluginInitializerContext) {
 const disabledDeprecation: ConfigDeprecation = (config, fromPath, addDeprecation) => {
   if (config.xpack?.spaces?.enabled === false) {
```

Comment:
What about changing this so we can completely remove the `enabled` option from the Spaces config?

```suggestion
  if ('enabled' in (settings?.xpack?.spaces || {}) {
```

Reply:
Hmm yeah maybe that's a good idea. I did leave it like this on purpose because I wanted to minimize the upgrade hurdle. But I forgot that it meant that we can't remove it from the config scheme 🤔 Personally I'm all for removing it, so if you're ok with that too, then I'll do it 😎